src/wp-includes/rest-api.php

```diff
@@ -1539,6 +1539,27 @@ function rest_stabilize_value( $value ) {
 	return $value;
 }
 
+/**
+ * Gets the schema of matching pattern property.
+ *
+ * @since 5.7.0
+ *
+ * @param string $property The property to check.
+ * @param array  $args     The schema array to use.
+ * @return array|null      The schema of matching pattern property.
+ */
+function rest_get_matching_pattern_property_schema( $property, $args ) {
+	if ( isset( $args['patternProperties'] ) ) {
+		foreach ( $args['patternProperties'] as $pattern => $child_schema ) {
+			if ( preg_match( '/' . $pattern . '/', $property ) ) {
```

Comment:
This should be updated to match how we handle the `pattern` keyword in `rest_validate_value_from_schema`. It might be worth extracting that into a reusable function.